### PR TITLE
fix(openai): streaming tool_call + logging multiple tool_call

### PR DESCRIPTION
**Pull request by guillaumefontan**
Fixes https://github.com/traceloop/openllmetry-js/issues/464 and https://github.com/traceloop/openllmetry-js/issues/465

### CODE REVIEW

File: packages/instrumentation-openai/src/instrumentation.ts [hunk 1]
```diff
@@ -388,27 +388,54 @@ export class OpenAIInstrumentation extends InstrumentationBase {
...
+            }
+            if (toolCall.function?.name) {
+              result.choices[0].message.tool_calls[toolCall.index].function.name += toolCall.function.name;
+            }
+            if (toolCall.function?.arguments) {
+              result.choices[0].message.tool_calls[toolCall.index].function.arguments += toolCall.function.arguments;
             }
           }
         }
+        // if (chunk.choices[0]?.delta.tool_calls) {
```
Comment (nirga) [inline packages/instrumentation-openai/src/instrumentation.ts]: nit: remove?

**Review by nirga — COMMENTED** [review]
Thanks! Left a small nit comment

File: packages/instrumentation-openai/src/instrumentation.ts (hunk 1 above)
Comment (guillaumefontan) [reply packages/instrumentation-openai/src/instrumentation.ts]: All done thanks!

**Comment by guillaumefontan** [comment]
@nirga just a note, this will break anyone's code which uses
`span.attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.function_call.name`]` (will return undefined)
as the new format will become
`span.attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.function_call.0.name`]`.
To avoid any breaking changes, we could make it so that the first tool call remains
`span.attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.function_call.name`]`
while all other tool calls are indexed:
`span.attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.function_call.1.name`]`.
However this may be confusing.

**Comment by nirga** [comment]
@guillaumefontan I think we should match the way we've implemented this on Python - 
https://github.com/traceloop/openllmetry/blob/a1b82d8b272ce3861d88f5a2898d3170aa9a6df0/packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/shared/chat_wrappers.py#L448

So it should be `tool_calls` (plural)

Also note the CI is broken on your PR

**Comment by guillaumefontan** [comment]
Hi @nirga I've updated to tool_calls and fixed the lint warnings. However the build error:
`Run pip3 install chromadb
error: externally-managed-environment`
does not appear to be related to my changes. Let me know if there is anything I can do.

**Review by nirga — APPROVED** [review]
Sorry for the delay here, working on fixing the CI now. LGTM except for that!

**Pull request merged**